### DISCUSSION TITLE
@pytest.mark.order(after) a parameterized function

### DISCUSSION THREAD

**Discussion by GijsPampiermole** [pytest-dev/pytest-order#136] [Q&A] [answered]
Hi, I've had a look around the internet, and tried lots of different things. Despite I cant seem to figure out how to use `order after` on a parameterized function, it always runs before the actual function runs.

Example:
```
tests/test.py::create    PASSED
tests/test.py::get[after=create]    PASSED
tests/test.py::get[after=update[bar]]    PASSED
tests/test.py::update[foo]    PASSED
tests/test.py::update[bar]    PASSED
```

Expected/Requested result:
```
tests/test.py::create    PASSED
tests/test.py::get[after=create]    PASSED
tests/test.py::update[foo]    PASSED
tests/test.py::update[bar]    PASSED
tests/test.py::get[after=update[bar]]    PASSED
```

**Comment by mrbean-bremen** [comment]
Parametrized functions are not fully supported (yet), see #55. That having said, can you show the example code that you want to order?

**Comment by GijsPampiermole** [comment]
This would be my current code, stripped down from everything
```python
def test_create():
    pass


@pytest.mark.order(after="test_create")
@pytest.mark.order(after="test_update[description]")
def test_get():
    pass


@pytest.mark.order(after="test_get")
@pytest.mark.parametrize("key", ["name"], ["description"])
def test_update(key):
    pass


@pytest.mark.order(after="test_get[test_get[description]]")
def test_delete():
    pass
  ```
  
  The order of tests I want to achieve is:
  - test_create
  - test_get
  - test_update[name]
  - test_update[description]
  - test_get
  - test_delete
  
  I hope this is enough information :D

**Reply by mrbean-bremen** [reply]
Is it correct that you want to run `test_get` twice?

**Reply by GijsPampiermole** [reply]
Yes, I want to run `test_get` twice. 
Once before updating the mutatable values(name, description) and once after updating, to ensure that none of the values corrupted the data.

**Reply by mrbean-bremen** [reply, marked as answer]
Ok, this is actually possible. What is not possible is to handle the parametrized tests separately (as you have tried), instead you just have to use the name without the id for the parametrized test:
```py
def test_create():
    pass


@pytest.mark.order(after="test_create")
@pytest.mark.order(after="test_update")
def test_get():
    pass


@pytest.mark.order(after="test_get[after=test_create]")
@pytest.mark.parametrize("key", ["name", "description"])
def test_update(key):
    pass


@pytest.mark.order(after="test_get[after=test_update]")
def test_delete():
    pass
```
I have to admit that I didn't know that tests that are running more than once are referenced this way - learned something new!

**Reply by GijsPampiermole** [reply]
Ah, I see, it works as intended now! Thanks a lot!.
I did not know I could reference the test without the `[param]`, as in the console it always show the test name with the param.

I guess we both learned something new! :D
Looking forward for the Parametrized support, and to continue using the tool!
